src/test/java/io/stargate/sgv2/it/DynamoApiTableIT.java

```diff
@@ -0,0 +1,96 @@
+import org.junit.jupiter.api.TestInstance;
+
+@QuarkusIntegrationTest
+@QuarkusTestResource(StargateTestResource.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DynamoApiTableIT extends DynamoITBase {
+  @Test
+  public void testDeleteNonExistentTable() {
+    DeleteTableRequest req = new DeleteTableRequest().withTableName("non-existent");
+    assertThrows(ResourceNotFoundException.class, () -> awsClient.deleteTable(req));
+    // TODO: ensure the exceptions are the same
+    assertThrows(Exception.class, () -> proxyClient.deleteTable(req));
+  }
+
+  @Test
+  public void testCreateThenDeleteTable() {
+    CreateTableRequest req =
+        new CreateTableRequest()
+            .withTableName("foo")
+            .withProvisionedThroughput(
+                new ProvisionedThroughput()
+                    .withReadCapacityUnits(100L)
+                    .withWriteCapacityUnits(100L))
+            .withKeySchema(new KeySchemaElement("Name", KeyType.HASH))
+            .withAttributeDefinitions(new AttributeDefinition("Name", ScalarAttributeType.S));
+
+    awsClient.createTable(req);
+    proxyClient.createTable(req);
+
+    awsClient.deleteTable("foo");
+    proxyClient.deleteTable("foo");
+  }
+
+  @Test
+  public void testListTables() {
+    CreateTableRequest req1 =
+        new CreateTableRequest()
+            .withTableName("aaa")
+            .withProvisionedThroughput(
+                new ProvisionedThroughput()
+                    .withReadCapacityUnits(100L)
+                    .withWriteCapacityUnits(100L))
+            .withKeySchema(new KeySchemaElement("Name", KeyType.HASH))
+            .withAttributeDefinitions(new AttributeDefinition("Name", ScalarAttributeType.S));
+    CreateTableRequest req2 = req1.clone().withTableName("bbb");
+    CreateTableRequest req3 = req1.clone().withTableName("ccc");
+    CreateTableRequest req4 = req1.clone().withTableName("ddd");
+
+    awsClient.createTable(req1);
+    awsClient.createTable(req2);
+    awsClient.createTable(req3);
+    awsClient.createTable(req4);
+    proxyClient.createTable(req1);
+    proxyClient.createTable(req2);
+    proxyClient.createTable(req3);
+    proxyClient.createTable(req4);
+
+    ListTablesResult awsResult;
+    ListTablesResult proxyResult;
+
+    // no args
+    awsResult = awsClient.listTables();
+    proxyResult = proxyClient.listTables();
+    assertEquals(awsResult, proxyResult);
+
+    // startTableName
+    awsResult = awsClient.listTables("bbb");
+    proxyResult = proxyClient.listTables("bbb");
+    assertEquals(awsResult, proxyResult);
+
+    // limit
+    awsResult = awsClient.listTables(2);
+    proxyResult = proxyClient.listTables(2);
+    assertEquals(awsResult, proxyResult);
+
+    // start + limit
+    awsResult = awsClient.listTables("bbb", 2);
+    proxyResult = proxyClient.listTables("bbb", 2);
+    assertEquals(awsResult, proxyResult);
+
+    // overflow
+    awsResult = awsClient.listTables(100);
```

Comment:
overflow test should use a number larger than 100, no?

Reply:
That's correct. My comment might creating confusion here. I originally want to test the case where (# of tables) < (# of query). But I didn't test the case where # of query > 100. Let me add that and revise this comment.